File: 2-sinif/4-yariyil/pdp/hft08/c_vscode/NesneYonelimliBenzetim2/src/Koleksiyon.c
#include "Koleksiyon.h"

Koleksiyon new_Koleksiyon(int* sayilar, int uzunluk){
	Koleksiyon this;
	this = (Koleksiyon)malloc(sizeof(struct KOLEKSIYON));
	this->super = new_Topluluk(sayilar, uzunluk);
	this->ortalama = &ortalama;
	this->yazdir = &yazdir_;
	this->delete_Koleksiyon = &delete_Koleksiyon;
	return this;
}
double ortalama(const Koleksiyon this){
	double toplam =0; 
	for(int i = 0; i < this->super->uzunluk; i++) {
		toplam += this->super->sayilar[i];
	}
	return toplam/this->super->uzunluk;
}
void yazdir_(const Koleksiyon this){
	this->super->yazdir(this->super);
	printf("\nOrtalama: %lf", ortalama(this));
}
void delete_Koleksiyon(const Koleksiyon this){
	if(this == NULL) return;
	this->super->delete_Topluluk(this->super);
	free(this);
}

File: 2-sinif/4-yariyil/pdp/hft08/java_eclipse/hft08/src/hft08/Koleksiyon.java
package hft08;

//ilk Topluluktan nesne oluşacak sonra Koleksiyondan
public class Koleksiyon extends Topluluk { //kalıtım alırken yukarı bilgi göndericek bir kurucu şarttır
	public Koleksiyon(int[] sayilar) {
		super(sayilar); //kurucu func'ın ilk satırında yukarı bilgi gönderilmelidir.
	}
	
	public double ortalama() {
		double toplam =0; 
		for(int sayi : sayilar) {
			toplam += sayi;
		}
		return toplam/sayilar.length; //protected olduğundan erişebildik
	}
	
	@Override
	public String toString() {
		return super.toString() + "\nOrtalama: " + ortalama();
	}
}
